src/components/application_manager/rpc_plugins/sdl_rpc_plugin/src/commands/mobile/register_app_interface_request.cc

```diff
@@ -1497,12 +1497,15 @@ void RegisterAppInterfaceRequest::CheckResponseVehicleTypeParam(
 void RegisterAppInterfaceRequest::SendSubscribeCustomButtonNotification() {
   using namespace smart_objects;
   using namespace hmi_apis;
+  LOG4CXX_AUTO_TRACE(logger_);
 
   SmartObject msg_params = SmartObject(SmartType_Map);
   msg_params[strings::app_id] = connection_key();
-  msg_params[strings::name] = Common_ButtonName::CUSTOM_BUTTON;
+  msg_params[strings::button_name] = Common_ButtonName::CUSTOM_BUTTON;
   msg_params[strings::is_suscribed] = true;
-  CreateHMINotification(FunctionID::Buttons_OnButtonSubscription, msg_params);
+
+  SendHMIRequest(
+      hmi_apis::FunctionID::Buttons_SubscribeButton, &msg_params, false);
```

Comment:
@mked-luxoft We are not waiting for HMI response here...What if HMI will not subscribe app for CUSTOM BUTTON?

Reply:
@AByzhynar, are you referring to handling events from HMI in `on_event` method? if so, i think that it will be handled in implementation of `SubscribeButtonRequest` class, since this request will be constructed and run as a result of call to `SendHMIRequest`

Reply:
@mked-luxoft Yes. But what code will be responsible as you explicitly stated that we are not going to subscribe for HMI response?

Reply:
@mked-luxoft Thanks. :+1: